scripts/find-documents-missing-metadata.ts

```diff
@@ -0,0 +1,152 @@
+const documentLimit = 10;
+
+const dryRun = true;
+
+const should_match_fields = [ "context_id", "key", "title", "type" ];
+const fields_to_copy = [ "unit", "investigation", "problem" ];
+
+const databaseURL = "https://collaborative-learning-ec215.firebaseio.com";
+
+// Fetch the service account key JSON file contents
+const credential = admin.credential.cert(getScriptRootFilePath("serviceAccountKey.json"));
+// Initialize the app with a service account, granting admin privileges
+admin.initializeApp({
+  credential,
+  databaseURL
+});
+
+console.log(`*** Starting to Scan Documents ***`);
+
+const collectionUrl = getFirestoreBasePath(portal, demo);
+
+const documentCollection = admin.firestore().collection(collectionUrl);
+
+let documentQuery = documentCollection
+  // Omit document types that are not expected to be associated with a unit, investigation, and problem
+  .where("type", "not-in", ["personal", "personalPublication", "learningLog", "learningLogPublication"])
+  .where("unit", "==", null);
+
+if (documentLimit) {
+  documentQuery = documentQuery.limit(documentLimit);
+}
+
+const singles = new Set();
+const triplets = new Set();
+const pairs = new Set();
+const mismatchDocs = new Set();
+const fixableDocsByType = { "problem": 0, "problemPublication": 0, "planning": 0 };
+
+const promises = [];
+
+const documentSnapshots = await documentQuery.get();
+documentSnapshots.forEach(doc => {
+  const data = doc.data();
+  // console.log(`Document ${doc.ref.path}: ${data.type} unit: ${data.unit}`);
+  // Look for other documents with the same key
+  const key = data.key;
+  const keyQuery = documentCollection.where("key", "==", key);
+  const query = keyQuery.get().then(async snapshot => {
+    if (snapshot.size === 1) {
+      singles.add(key);
+    } else if (snapshot.size > 2) {
+      triplets.add(key);
+    } else if (snapshot.size === 2) {
+      if (!pairs.has(key)) {
+        pairs.add(key);
+        console.log(`Found ${snapshot.size} documents with key ${key}`);
+        console.log(`  ${snapshot.docs[0].ref.path} and ${snapshot.docs[1].ref.path}`);
+        let goodDoc, badDoc;
+        let badRef: admin.firestore.QueryDocumentSnapshot<admin.firestore.DocumentData>;
+        if (snapshot.docs[0].data().unit) {
+          goodDoc = snapshot.docs[0].data();
+          badDoc = snapshot.docs[1].data();
+          badRef = snapshot.docs[1];
+        } else if (snapshot.docs[1].data().unit) {
+          goodDoc = snapshot.docs[1].data();
+          badDoc = snapshot.docs[0].data();
+          badRef = snapshot.docs[0];
+        } else {
+          console.log(`   Neither document has metadata`);
+        }
+        if (goodDoc) {
+          // console.log("good doc", goodDoc);
+          // Make sure the main fields match in the two documents
+          let match = true;
+          should_match_fields.forEach(field => {
+            // Treat undefined and null as equivalent
+            const f1 = goodDoc[field] || null;
```

Comment:
Could the `title` field be an empty string ever and thus be "falsey"?  If so this should probably be:

`goodDoc[field] ?? null`

to check for undefined/null only.